connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java

```diff
@@ -86,18 +86,40 @@ public ConnectorsResource(Herder herder) {
     @POST
     @Path("/")
     public Response createConnector(final @QueryParam("forward") Boolean forward,
-                                    final CreateConnectorRequest createRequest) throws Throwable {
-        String name = createRequest.name();
-        if (name.contains("/")) {
-            throw new BadRequestException("connector name should not contain '/'");
-        }
+                                     final CreateConnectorRequest createRequest) throws Throwable {
         Map<String, String> configs = createRequest.config();
-        if (!configs.containsKey(ConnectorConfig.NAME_CONFIG))
-            configs.put(ConnectorConfig.NAME_CONFIG, name);
+        CreateConnectorRequest finalCreateRequest = createRequest;
+
+        String nameFromBody = createRequest.name();
+        String nameFromConfig = createRequest.config().get(ConnectorConfig.NAME_CONFIG);
+        String name = "";
+
+        if (nameFromBody == null && nameFromConfig == null) {
```

Comment:
Those tests might be due to a misunderstanding on my part, back when I looked at this, it was possible to send a request without a top level "name" object, so I figured that it was legal to specify the name within the config object of the request. The current version does not accept requests without the top level "name" element, so I'll remove these tests.

I put them here, because there could potentially have been two connector names and we can only pass one along to the config object, so they were not possible to do in the validator.